ui/apps/platform/src/Containers/NetworkGraph/types/topology.type.ts

```diff
@@ -0,0 +1,69 @@
+import { Model, NodeModel } from '@patternfly/react-topology';
+
+import { ListenPort } from 'types/networkFlow.proto';
+import { Override } from 'utils/type.utils';
+
+export type CustomModel = Override<Model, { nodes?: CustomNodeModel[] }>;
+
+export type CustomNodeModel =
+    | NamespaceNodeModel
+    | DeploymentNodeModel
+    | ExternalNodeModel
+    | ExternalEntitiesNodeModel
+    | CIDRBlockNodeModel;
+
+export type NamespaceNodeModel = Override<NodeModel, { data: NamespaceData }>;
+
+export type DeploymentNodeModel = Override<NodeModel, { data: DeploymentData }>;
+
+export type ExternalNodeModel = Override<NodeModel, { data: ExternalData }>;
+
+export type ExternalEntitiesNodeModel = Override<NodeModel, { data: ExternalEntitiesData }>;
+
+export type CIDRBlockNodeModel = Override<NodeModel, { data: CIDRBlockData }>;
+
+export type CustomNodeData =
+    | NamespaceData
+    | DeploymentData
+    | ExternalData
+    | ExternalEntitiesData
+    | CIDRBlockData;
+
+export type NamespaceData = {
+    type: 'NAMESPACE';
+    collapsible: boolean;
+    showContextMenu: boolean;
+};
+
+export type DeploymentData = {
+    type: 'DEPLOYMENT';
+    id: string;
+    deployment: {
+        cluster: string;
+        listenPorts: ListenPort[];
+        name: string;
+        namespace: string;
+    };
+    policyIds: string[];
```

Comment:
i think you can use the `DeploymentNetworkEntityInfo` type here

Reply:
Yea true, I can do that. We have those types defined in `networkFlow.proto.ts`. The thing is that I do want to keep the types different. For example, in that `networkFlow.proto.ts` we use the type `INTERNET` but I would want to use `EXTERNAL_ENTITIES`. Since the deployment is pretty much the same, I could use the already existing type, but then it would basically be the only one using the existing type. I decided not to use it because it seemed cleaner and simpler to just define it here instead.